When using reactive API,  Lettuce cannot handle redis response. Errors java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: io.lettuce.core.output.BooleanListOutput does not support set(boolean)

## Bug Report

<!-- First of all: Have you checked the docs https://github.com/lettuce-io/lettuce-core/wiki, GitHub issues, or Stack Overflow whether someone else has already reported your issue? -->

#### Current Behavior
Using Redis in docker on localhost. Issuing command via  extending Commands interaface.
Commands.

@Command("BF.INSERT :filter CAPACITY :capacity ERROR :error ITEMS :items ")
    Flux<Boolean>  insert(@Param("filter") String filter, @Param("capacity") long capacity,
                                @Param("error") double error, @Param("item") String[] items);
                               
                                when I convert the result of any command issued;
                                Flux<Boolean> res= bloomCommand.insert("myBloomFilter", 10000, 0.01, idsArray). 
                                
When I collect this value or convert to futures I get this exception. Errors java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: io.lettuce.core.output.BooleanListOutput does not support set(boolean)

 . Redis version 7.2.x 
Lettuce Core 6.3.1
SpringBoot - 2.5.12
<!-- A clear and concise description of the behavior.-->

However when I use Mono :
Mono<Boolean>  insert(@Param("filter") String filter, @Param("capacity") long capacity,
                                @Param("error") double error, @Param("item") String[] item);
                                
                                the response is decoded correctly. 

I also tried dispatch but for that to work I had to  overload  set(boolean) method.
CommandArgs<String, String> args = new CommandArgs<>(StringCodec.UTF8)
                .addKey("myBloomFilter")
                .add("CAPACITY").add(10000)
                .add("ERROR").add(0.01)
                .add("ITEMS").add("item1").add("item2").add("item3");

class BFInsert implements ProtocolKeyword {
    @Override
    public byte[] getBytes() {
        return "BF.INSERT".getBytes();
    }

    @Override
    public String name() {
        return "Nits";
    }
}

        List<Boolean> response = syncCommands.dispatch(new BFInsert(),
                new BooleanListOutput(StringCodec.UTF8),
                args);

        System.out.println("Response: " + response);



class BooleanListOutput extends CommandOutput<String, String, List<Boolean>> {
    private final List<Boolean> booleans = new ArrayList<>();

    public BooleanListOutput(RedisCodec<String, String> codec) {
        super(codec, new ArrayList<>());
    }

    @Override
    public void set(boolean value) {
        booleans.add(value);
    }

    @Override
    public void complete(int depth) {
        this.output = booleans;
    }
}

<details>
<summary>Stack trace</summary>

reactor.core.Exceptions$ErrorCallbackNotImplemented: java.lang.UnsupportedOperationException: io.lettuce.core.output.BooleanListOutput does not support set(boolean)
Caused by: java.lang.UnsupportedOperationException: io.lettuce.core.output.BooleanListOutput does not support set(boolean)
	at io.lettuce.core.output.CommandOutput.set(CommandOutput.java:129)
	at io.lettuce.core.protocol.RedisStateMachine.safeSet(RedisStateMachine.java:758)
	at io.lettuce.core.protocol.RedisStateMachine.handleBoolean(RedisStateMachine.java:419)
	at io.lettuce.core.protocol.RedisStateMachine$State$Type.handle(RedisStateMachine.java:205)
	at io.lettuce.core.protocol.RedisStateMachine.doDecode(RedisStateMachine.java:339)
	at io.lettuce.core.protocol.RedisStateMachine.decode(RedisStateMachine.java:300)
	at io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:840)
	at io.lettuce.core.protocol.CommandHandler.decode0(CommandHandler.java:791)
	at io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:765)
	at io.lettuce.core.protocol.CommandHandler.decode(CommandHandler.java:657)
	at io.lettuce.core.protocol.CommandHandler.channelRead(CommandHandler.java:597)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

```java
// your stack trace here;
```

</details>

#### Input Code

<!-- Java/Kotlin/Scala/Groovy/… or Repo link if applicable: -->

<details>
<summary>Input Code</summary>

```java
// your code here;
```
        RedisCommandFactory commandFactory = new RedisCommandFactory(redisClient.sconnect());
        bloomCommand = commandFactory.getCommands(BloomCommandReactive.class);

</details>

#### Expected behavior/code

<!-- A clear and concise description of what you expected to happen (or code). -->

#### Environment

- Lettuce version(s): [6.3.1.RELEASE]
- Redis version: [7.2.x]

#### Possible Solution

<!-- Only if you have suggestions on a fix for the bug -->

#### Additional context

<!-- Add any other context about the problem here. Do not add code as screenshots. -->

--- a/src/main/java/io/lettuce/core/output/BooleanListOutput.java
+++ b/src/main/java/io/lettuce/core/output/BooleanListOutput.java
@@ -50,6 +50,11 @@ public class BooleanListOutput<K, V> extends CommandOutput<K, V, List<Boolean>> 
     }
 
     @Override
+    public void set(boolean value) {
+        subscriber.onNext(output, value);
+    }
+
+    @Override
     public void multi(int count) {
 
         if (!initialized) {

--- a/src/test/java/io/lettuce/core/commands/CommandInterfacesIntegrationTests.java
+++ b/src/test/java/io/lettuce/core/commands/CommandInterfacesIntegrationTests.java
@@ -1,0 +1,110 @@
+/*
+ * Copyright 2011-Present, Redis Ltd. and Contributors
+ * All rights reserved.
+ *
+ * Licensed under the MIT License.
+ *
+ * This file contains contributions from third-party contributors
+ * licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lettuce.core.commands;
+
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisContainerIntegrationTests;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.TestSupport;
+import io.lettuce.core.api.StatefulConnection;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.dynamic.Commands;
+import io.lettuce.core.dynamic.RedisCommandFactory;
+import io.lettuce.core.dynamic.annotation.Command;
+import io.lettuce.core.dynamic.annotation.Param;
+import io.lettuce.test.LettuceExtension;
+import io.lettuce.test.condition.EnabledOnCommand;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import reactor.core.publisher.Flux;
+
+import javax.inject.Inject;
+import java.lang.reflect.Proxy;
+import java.util.List;
+
+import static io.lettuce.TestTags.INTEGRATION_TEST;
+import static io.lettuce.core.SetArgs.Builder.ex;
+import static io.lettuce.core.SetArgs.Builder.exAt;
+import static io.lettuce.core.SetArgs.Builder.px;
+import static io.lettuce.core.SetArgs.Builder.pxAt;
+import static io.lettuce.core.StringMatchResult.Position;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Integration tests for {@link io.lettuce.core.dynamic.annotation.Command}.
+ *
+ * @author Tihomir Mateev
+ */
+@Tag(INTEGRATION_TEST)
+public class CommandInterfacesIntegrationTests extends RedisContainerIntegrationTests {
+
+    protected static RedisClient client;
+
+    protected static RedisCommands<String, String> redis;
+
+    protected CommandInterfacesIntegrationTests() {
+        RedisURI redisURI = RedisURI.Builder.redis("127.0.0.1").withPort(16379).build();
+
+        client = RedisClient.create(redisURI);
+        redis = client.connect().sync();
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.redis.flushall();
+    }
+
+    @Test
+    void issue2612() {
+        CustomCommands commands = CustomCommands.instance(getConnection());
+
+        Flux<Boolean> flux = commands.insert("myBloomFilter", 10000, 0.01, new String[] { "1", "2", "3", });
+        List<Boolean> res = flux.collectList().block();
+
+        assertThat(res).hasSize(3);
+        assertThat(res).contains(true, true, true);
+    }
+
+    protected StatefulConnection<String, String> getConnection() {
+        StatefulRedisConnection<String, String> src = redis.getStatefulConnection();
+        Assumptions.assumeFalse(Proxy.isProxyClass(src.getClass()), "Redis connection is proxy, skipping.");
+        return src;
+    }
+
+    private interface CustomCommands extends Commands {
+
+        @Command("BF.INSERT :filter CAPACITY :capacity ERROR :error ITEMS :items ")
+        Flux<Boolean> insert(@Param("filter") String filter, @Param("capacity") long capacity, @Param("error") double error,
+                @Param("items") String[] items);
+
+        static CustomCommands instance(StatefulConnection<String, String> conn) {
+            RedisCommandFactory factory = new RedisCommandFactory(conn);
+            return factory.getCommands(CustomCommands.class);
+        }
+
+    }
+
+}

